src/components/TipControl.vue

```diff
@@ -0,0 +1,205 @@
+
+<template>
+  <div
+    class="tip-url__wrapper"
+    :title="isTipped ? 'Total tips (you tipped too)' : 'Total tips (click to retip the same URL)'"
+  >
+    <div
+      v-if="show"
+      class="overlay"
+      @click="toggleTip(false)"
+    />
+    <div
+      class="tip__container_wrapper"
+      @:click.stop
+    >
+      <div
+        class="retip__content"
+        :class="[{ active: show }]"
+        @click="toggleTip(!show)"
+      >
+        <img
+          class="tip__icon"
+          :src="iconTip"
+        >
+        <ae-amount
+          :amount="tip.total_amount"
+          :round="2"
+        />
+        <fiat-value :amount="tip.total_amount.toString()" />
+      </div>
+      <div
+        v-if="show"
+        class="tip__container"
+      >
+        <loading :show-loading="showLoading" />
+        <div
+          v-show="error && !showLoading"
+          class="text-center mb-2"
+        >
+          An error occurred while sending tip
+        </div>
+        <div
+          v-if="!showLoading"
+          class="input-group"
+        >
+          <input
+            v-model="message"
+            type="text"
+            class="form-control tip__message"
+            placeholder="Add message"
+          >
+        </div>
+        <div v-if="!showLoading">
```

Comment:
I would have made that a `<form>` with `@onsubmit="submitAction()` and put the `<ae-button>` to render a `button type=submit`. This way pressing Enter will work also will be somehow better supported on mobile